src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,251 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO.Abstractions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Platform.Metrics.LinuxConsumption;
+using Microsoft.Azure.WebJobs.Script.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Metrics
+{
+    public class LinuxContainerLegionMetricsPublisher : IMetricsPublisher, IDisposable
+    {
+        private readonly ILinuxConsumptionMetricsTracker _metricsTracker;
+        private readonly LegionMetricsFileManager _metricsFileManager;
+        private readonly TimeSpan _memorySnapshotInterval = TimeSpan.FromMilliseconds(1000);
+        private readonly TimeSpan _timerStartDelay = TimeSpan.FromSeconds(2);
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private readonly IDisposable _standbyOptionsOnChangeSubscription;
+        private readonly IEnvironment _environment;
+        private readonly ILogger _logger;
+        private readonly IMetricsLogger _metricsLogger;
+        private readonly string _containerName;
+
+        private TimeSpan _metricPublishInterval;
+        private Process _process;
+        private Timer _processMonitorTimer;
+        private Timer _metricsPublisherTimer;
+        private bool _initialized = false;
+
+        public LinuxContainerLegionMetricsPublisher(IEnvironment environment, IOptionsMonitor<StandbyOptions> standbyOptions, ILogger<LinuxContainerLegionMetricsPublisher> logger, IFileSystem fileSystem, ILinuxConsumptionMetricsTracker metricsTracker, IScriptHostManager scriptHostManager, int? metricsPublishIntervalMS = null)
+        {
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _metricsTracker = metricsTracker ?? throw new ArgumentNullException(nameof(metricsTracker));
+            if (scriptHostManager == null || !Utility.TryGetHostService<IMetricsLogger>(scriptHostManager, out _metricsLogger))
+            {
+                throw new InvalidOperationException("Unable to get IMetricsLogger Service.");
+            }
+            _containerName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.ContainerName);
```

Comment:
Is there no options object with container name on it? Or can we create one? You can combine all different env vars you are retrieving onto it (such as metrics file path).

Reply:
Added.